sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -296,6 +338,11 @@ def body_as_str(self, encoding="UTF-8"):
         """
         data = self.body
         try:
+            if self.body_type != AmqpMessageBodyType.DATA:
+                # pylint: disable=protected-access
+                self.raw_amqp_message._message._body._encoding = encoding
+                return str(self.raw_amqp_message._message._body)
```

Comment:
@Adam should I save the encoding here and restore it after calling str()?
```suggestion
                old_encoding = self.raw_amqp_message._message._body._encoding
                self.raw_amqp_message._message._body._encoding = encoding
                body_str = str(self.raw_amqp_message._message._body)
                self.raw_amqp_message._message._body._encoding = old_encoding
                return body_str
```